diff --git a/firmwares/bcm4366c0/structs.common.h b/firmwares/bcm4366c0/structs.common.h
index 9de7c73..07264d0 100644
--- a/firmwares/bcm4366c0/structs.common.h
+++ b/firmwares/bcm4366c0/structs.common.h
@@ -1077,6 +1077,18 @@ struct d11regs {
 	} u;
 } __attribute__((packed));
 
+struct wlcband {
+	int	bandtype;
+	int	bandunit;
+	short	phytype;
+	short	phyrev;
+	short	radioid;
+	short	radiorev;
+	void	*res;
+	char	gmode;
+	void	*hwrs_scb;
+};
+
 struct wlc_info {
     void *pub;                  /* 0x00 */
     struct osl_info *osh;       /* 0x04 */
@@ -1089,7 +1101,7 @@ struct wlc_info {
     int PAD;                    /* 0x20 */
     int PAD;                    /* 0x24 */
     int PAD;                    /* 0x28 */
-    int PAD;                    /* 0x2c */
+    struct wlcband *band;       /* 0x2c */
     int PAD;                    /* 0x30 */
     int PAD;                    /* 0x34 */
     int PAD;                    /* 0x38 */
@@ -1181,6 +1193,248 @@ struct wlc_info {
     int PAD;                    /* 0x190 */
     int PAD;                    /* 0x194 */
     void *scan;                 /* 0x198 */
+    int PAD;                    /* 0x19c */
+    int PAD;                    /* 0x1a0 */
+    int PAD;                    /* 0x1a4 */
+    int PAD;                    /* 0x1a8 */
+    int PAD;                    /* 0x1ac */
+    int PAD;                    /* 0x1b0 */
+    int PAD;                    /* 0x1b4 */
+    int PAD;                    /* 0x1b8 */
+    int PAD;                    /* 0x1bc */
+    int PAD;                    /* 0x1c0 */
+    int PAD;                    /* 0x1c4 */
+    int PAD;                    /* 0x1c8 */
+    int PAD;                    /* 0x1cc */
+    int PAD;                    /* 0x1d0 */
+    int PAD;                    /* 0x1d4 */
+    int PAD;                    /* 0x1d8 */
+    int PAD;                    /* 0x1dc */
+    int PAD;                    /* 0x1e0 */
+    int PAD;                    /* 0x1e4 */
+    int PAD;                    /* 0x1e8 */
+    int PAD;                    /* 0x1ec */
+    int PAD;                    /* 0x1f0 */
+    int PAD;                    /* 0x1f4 */
+    int PAD;                    /* 0x1f8 */
+    int PAD;                    /* 0x1fc */
+    int PAD;                    /* 0x200 */
+    int PAD;                    /* 0x204 */
+    int PAD;                    /* 0x208 */
+    int PAD;                    /* 0x20c */
+    int PAD;                    /* 0x210 */
+    int PAD;                    /* 0x214 */
+    int PAD;                    /* 0x218 */
+    int PAD;                    /* 0x21c */
+    int PAD;                    /* 0x220 */
+    int PAD;                    /* 0x224 */
+    int PAD;                    /* 0x228 */
+    int PAD;                    /* 0x22c */
+    int PAD;                    /* 0x230 */
+    int PAD;                    /* 0x234 */
+    int PAD;                    /* 0x238 */
+    int PAD;                    /* 0x23c */
+    int PAD;                    /* 0x240 */
+    int PAD;                    /* 0x244 */
+    int PAD;                    /* 0x248 */
+    int PAD;                    /* 0x24c */
+    int PAD;                    /* 0x250 */
+    int PAD;                    /* 0x254 */
+    int PAD;                    /* 0x258 */
+    int PAD;                    /* 0x25c */
+    int PAD;                    /* 0x260 */
+    int PAD;                    /* 0x264 */
+    int PAD;                    /* 0x268 */
+    int PAD;                    /* 0x26c */
+    int PAD;                    /* 0x270 */
+    int PAD;                    /* 0x274 */
+    int PAD;                    /* 0x278 */
+    int PAD;                    /* 0x27c */
+    int PAD;                    /* 0x280 */
+    int PAD;                    /* 0x284 */
+    int PAD;                    /* 0x288 */
+    int PAD;                    /* 0x28c */
+    int PAD;                    /* 0x290 */
+    int PAD;                    /* 0x294 */
+    int PAD;                    /* 0x298 */
+    int PAD;                    /* 0x29c */
+    int PAD;                    /* 0x2a0 */
+    int PAD;                    /* 0x2a4 */
+    int PAD;                    /* 0x2a8 */
+    int PAD;                    /* 0x2ac */
+    int PAD;                    /* 0x2b0 */
+    int PAD;                    /* 0x2b4 */
+    int PAD;                    /* 0x2b8 */
+    int PAD;                    /* 0x2bc */
+    int PAD;                    /* 0x2c0 */
+    int PAD;                    /* 0x2c4 */
+    int PAD;                    /* 0x2c8 */
+    int PAD;                    /* 0x2cc */
+    int PAD;                    /* 0x2d0 */
+    int PAD;                    /* 0x2d4 */
+    int PAD;                    /* 0x2d8 */
+    int PAD;                    /* 0x2dc */
+    int PAD;                    /* 0x2e0 */
+    int PAD;                    /* 0x2e4 */
+    int PAD;                    /* 0x2e8 */
+    int PAD;                    /* 0x2ec */
+    int PAD;                    /* 0x2f0 */
+    int PAD;                    /* 0x2f4 */
+    int PAD;                    /* 0x2f8 */
+    int PAD;                    /* 0x2fc */
+    int PAD;                    /* 0x300 */
+    int PAD;                    /* 0x304 */
+    int PAD;                    /* 0x308 */
+    int PAD;                    /* 0x30c */
+    int PAD;                    /* 0x310 */
+    int PAD;                    /* 0x314 */
+    int PAD;                    /* 0x318 */
+    int PAD;                    /* 0x31c */
+    int PAD;                    /* 0x320 */
+    int PAD;                    /* 0x324 */
+    int PAD;                    /* 0x328 */
+    int PAD;                    /* 0x32c */
+    int PAD;                    /* 0x330 */
+    int PAD;                    /* 0x334 */
+    int PAD;                    /* 0x338 */
+    int PAD;                    /* 0x33c */
+    int PAD;                    /* 0x340 */
+    int PAD;                    /* 0x344 */
+    int PAD;                    /* 0x348 */
+    int PAD;                    /* 0x34c */
+    int PAD;                    /* 0x350 */
+    int PAD;                    /* 0x354 */
+    int PAD;                    /* 0x358 */
+    int PAD;                    /* 0x35c */
+    int PAD;                    /* 0x360 */
+    int PAD;                    /* 0x364 */
+    int PAD;                    /* 0x368 */
+    int PAD;                    /* 0x36c */
+    int PAD;                    /* 0x370 */
+    int PAD;                    /* 0x374 */
+    int PAD;                    /* 0x378 */
+    int PAD;                    /* 0x37c */
+    int PAD;                    /* 0x380 */
+    int PAD;                    /* 0x384 */
+    int PAD;                    /* 0x388 */
+    int PAD;                    /* 0x38c */
+    int PAD;                    /* 0x390 */
+    int PAD;                    /* 0x394 */
+    int PAD;                    /* 0x398 */
+    int PAD;                    /* 0x39c */
+    int PAD;                    /* 0x3a0 */
+    int PAD;                    /* 0x3a4 */
+    int PAD;                    /* 0x3a8 */
+    int PAD;                    /* 0x3ac */
+    int PAD;                    /* 0x3b0 */
+    int PAD;                    /* 0x3b4 */
+    int PAD;                    /* 0x3b8 */
+    int PAD;                    /* 0x3bc */
+    int PAD;                    /* 0x3c0 */
+    int PAD;                    /* 0x3c4 */
+    int PAD;                    /* 0x3c8 */
+    int PAD;                    /* 0x3cc */
+    int PAD;                    /* 0x3d0 */
+    int PAD;                    /* 0x3d4 */
+    int PAD;                    /* 0x3d8 */
+    int PAD;                    /* 0x3dc */
+    int PAD;                    /* 0x3e0 */
+    int PAD;                    /* 0x3e4 */
+    int PAD;                    /* 0x3e8 */
+    int PAD;                    /* 0x3ec */
+    int PAD;                    /* 0x3f0 */
+    int PAD;                    /* 0x3f4 */
+    int PAD;                    /* 0x3f8 */
+    int PAD;                    /* 0x3fc */
+    int PAD;                    /* 0x400 */
+    int PAD;                    /* 0x404 */
+    int PAD;                    /* 0x408 */
+    int PAD;                    /* 0x40c */
+    int PAD;                    /* 0x410 */
+    int PAD;                    /* 0x414 */
+    int PAD;                    /* 0x418 */
+    int PAD;                    /* 0x41c */
+    int PAD;                    /* 0x420 */
+    int PAD;                    /* 0x424 */
+    int PAD;                    /* 0x428 */
+    int PAD;                    /* 0x42c */
+    int PAD;                    /* 0x430 */
+    int PAD;                    /* 0x434 */
+    int PAD;                    /* 0x438 */
+    int PAD;                    /* 0x43c */
+    int PAD;                    /* 0x440 */
+    int PAD;                    /* 0x444 */
+    int PAD;                    /* 0x448 */
+    int PAD;                    /* 0x44c */
+    int PAD;                    /* 0x450 */
+    int PAD;                    /* 0x454 */
+    int PAD;                    /* 0x458 */
+    int PAD;                    /* 0x45c */
+    int PAD;                    /* 0x460 */
+    int PAD;                    /* 0x464 */
+    int PAD;                    /* 0x468 */
+    int PAD;                    /* 0x46c */
+    int PAD;                    /* 0x470 */
+    int PAD;                    /* 0x474 */
+    int PAD;                    /* 0x478 */
+    int PAD;                    /* 0x47c */
+    int PAD;                    /* 0x480 */
+    int PAD;                    /* 0x484 */
+    int PAD;                    /* 0x488 */
+    int PAD;                    /* 0x48c */
+    int PAD;                    /* 0x490 */
+    int PAD;                    /* 0x494 */
+    int PAD;                    /* 0x498 */
+    int PAD;                    /* 0x49c */
+    int PAD;                    /* 0x4a0 */
+    int PAD;                    /* 0x4a4 */
+    int PAD;                    /* 0x4a8 */
+    int PAD;                    /* 0x4ac */
+    int PAD;                    /* 0x4b0 */
+    int PAD;                    /* 0x4b4 */
+    int PAD;                    /* 0x4b8 */
+    int PAD;                    /* 0x4bc */
+    int PAD;                    /* 0x4c0 */
+    int PAD;                    /* 0x4c4 */
+    int PAD;                    /* 0x4c8 */
+    int PAD;                    /* 0x4cc */
+    int PAD;                    /* 0x4d0 */
+    int PAD;                    /* 0x4d4 */
+    int PAD;                    /* 0x4d8 */
+    int PAD;                    /* 0x4dc */
+    int PAD;                    /* 0x4e0 */
+    int PAD;                    /* 0x4e4 */
+    int PAD;                    /* 0x4e8 */
+    int PAD;                    /* 0x4ec */
+    int PAD;                    /* 0x4f0 */
+    int PAD;                    /* 0x4f4 */
+    int PAD;                    /* 0x4f8 */
+    int PAD;                    /* 0x4fc */
+    int PAD;                    /* 0x500 */
+    int PAD;                    /* 0x504 */
+    int PAD;                    /* 0x508 */
+    int PAD;                    /* 0x50c */
+    int PAD;                    /* 0x510 */
+    int PAD;                    /* 0x514 */
+    int PAD;                    /* 0x518 */
+    int PAD;                    /* 0x51c */
+    int PAD;                    /* 0x520 */
+    int PAD;                    /* 0x524 */
+    int PAD;                    /* 0x528 */
+    int PAD;                    /* 0x52c */
+    int PAD;                    /* 0x530 */
+    int PAD;                    /* 0x534 */
+    int PAD;                    /* 0x538 */
+    int PAD;                    /* 0x53c */
+    int PAD;                    /* 0x540 */
+    int PAD;                    /* 0x544 */
+    int PAD;                    /* 0x548 */
+    int PAD;                    /* 0x54c */
+    int PAD;                    /* 0x550 */
+    int PAD;                    /* 0x554 */
+    int PAD;                    /* 0x558 */
+    int PAD;                    /* 0x55c */
+    void *active_queue;         /* 0x560 */
 };
 
 typedef void (*to_fun_t)(void *arg);
diff --git a/patches/common/wrapper.c b/patches/common/wrapper.c
index e098256..09cdb52 100644
--- a/patches/common/wrapper.c
+++ b/patches/common/wrapper.c
@@ -778,6 +778,7 @@ AT(CHIP_VER_BCM43596a0, FW_VER_ALL, 0x470cc)
 AT(CHIP_VER_BCM43455, FW_VER_ALL, 0x31CE8)
 AT(CHIP_VER_BCM43455c0, FW_VER_ALL, 0x31CE8)
 AT(CHIP_VER_BCM43436b0, FW_VER_9_88_4_65, 0xbd4c)
+AT(CHIP_VER_BCM4366c0, FW_VER_10_10_122_20, 0x230f54)
 int
 wlc_sendctl(void *wlc, void *p, void *qi, void *scb, unsigned int fifo, unsigned int rate_override, char enq_only)
 RETURN_DUMMY
